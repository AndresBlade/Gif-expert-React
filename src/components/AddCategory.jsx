import PropTypes from 'prop-types'
import { useState } from "react"

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = e => {
        e.preventDefault()
        if (inputValue.trim().length <= 1) return
        onNewCategory(inputValue.trim())
        setInputValue('')
    }

    return (
        <form name='gif-search' onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}