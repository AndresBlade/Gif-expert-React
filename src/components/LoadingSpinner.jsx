import React from 'react'
import '../spinner.css'

export const LoadingSpinner = ({ isLoading }) => {
    return (
        isLoading && (<div className="spinner-container" data-testid='spinner'>
            <div className="spinner"></div>
        </div>)
    )
}
