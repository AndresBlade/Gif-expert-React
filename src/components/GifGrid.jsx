import PropTypes from "prop-types"
import { GifItem } from "./GifItem"
import { useFetchGifs } from '../hooks/useFetchGifs'
import { LoadingSpinner } from "./LoadingSpinner"

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category)

    // const [images, setImages] = useState([])

    // const getImages = async () => {
    //     const newImages = await getGifs(category)
    //     setImages(newImages)
    // }

    // useEffect(() => {
    //     getImages()
    // }, [])

    return (
        <>
            <LoadingSpinner isLoading={isLoading} />
            <div className="card-grid">
                {
                    images.map((image) =>
                        <GifItem key={image.id} {...image} />
                    )}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}


