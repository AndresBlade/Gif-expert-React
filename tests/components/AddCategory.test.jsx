import { fireEvent, render, screen } from '@testing-library/react'
import { AddCategory } from '../../src/components/AddCategory'
import UserEvent from '@testing-library/user-event'


describe('Pruebas en <AddCategory />', () => {
    const inputValue = 'Prueba'

    test('Debería de hacer match con el snapshot anterior', () => {

        const { container } = render(<AddCategory onNewCategory={() => { }} />)
        expect(container).toMatchSnapshot()
    })

    test('Debería cambiar el valor de la caja de texto', async () => {

        render(<AddCategory onNewCategory={() => { }} />)
        const user = UserEvent.setup()
        const input = screen.getByRole('textbox');
        await user.type(input, inputValue)
        expect(input).toHaveValue(inputValue)
    })

    test('Debería de llamar onNewCategory si el input tiene un valor', async () => {
        const user = UserEvent.setup()
        const onNewCategory = jest.fn()
        //ToDo
        render(<AddCategory onNewCategory={onNewCategory} />)
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form')
        await user.type(input, inputValue)
        fireEvent.submit(form)
        expect(input).toHaveValue('')

        expect(onNewCategory).toBeCalledTimes(1)
        expect(onNewCategory).toBeCalledWith(inputValue)




    })

    test('No debe de llamar onNewCategory si el input esta vacio', async () => {
        const user = UserEvent.setup()
        const onNewCategory = jest.fn()
        //ToDo
        render(<AddCategory onNewCategory={onNewCategory} />)
        const form = screen.getByRole('form')
        fireEvent.submit(form)

        expect(onNewCategory).not.toBeCalled()
    })

})