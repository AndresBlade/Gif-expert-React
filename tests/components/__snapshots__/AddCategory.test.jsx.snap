// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Deber√≠a de hacer match con el snapshot anterior 1`] = `
<div>
  <form
    name="gif-search"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
